Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ReviewRequestController.java

```diff
@@ -21,12 +22,33 @@
            
     @RequestMapping(value = "/auth/request/add", method = RequestMethod.POST )
     public @ResponseBody ReviewRequest addNewReviewRequest( @ModelAttribute(value="reviewRequest") ReviewRequest reviewRequest,Principal principal) {
-  
-        reviewRequest.setStatus(ReviewStatus.OPENED);
         reviewRequest.setRequester(principal.getName());
         
         requestsMgr.addReviewRequest(reviewRequest);
         
         return reviewRequest;
     }
+    
+    @RequestMapping(value = "/auth/request/resolve", method = RequestMethod.POST )
+    public @ResponseBody ReviewRequest resolveRequest(@ModelAttribute(value="reviewRequest") ReviewRequest reviewRequest,Principal principal) {
+        reviewRequest.setResolver(principal.getName());
+        
+        ReviewRequest updatedRequest = requestsMgr.updateReview(reviewRequest);
+        
+        return updatedRequest;
+    }
+    
+    @RequestMapping(value = "/auth/request/reopen", method = RequestMethod.POST )
+    public @ResponseBody ReviewRequest reopenRequest(@ModelAttribute(value="reviewRequest") ReviewRequest reviewRequest,Principal principal) {
+        ReviewRequest updatedRequest = requestsMgr.reopenReview(reviewRequest);
+        
+        return updatedRequest;
+    }
+    
+    @RequestMapping(value = "/auth/request/getallreviews/{id}", method = RequestMethod.GET)
```

Comment:
make that just ```/auth/requests/{id}/all```

Reply:
also, I would call ```id``` ```conceptId```, it is the concept id, right? id sounds like the request id.